Add support for custom timestamps in chat and logs.

louipc opened the pull request:
I increased the size of TIME_STR_SIZE to accomodate log timestamps.
Also have the old 24/12 format deprecated, but it might just make sense to have it removed altogether.

louipc commented:
Seems I will have to sanitize the timestamp or change the name of the groupchat log somehow. I guess this will be a platform specific thing. Log filename can't have '/' on *nix, or '\' on windows I imagine. What could be a sane way to deal with this?

louipc commented:
Ok I just reverted the change I made to the groupchat log filename timestamp. I guess that can be changed at some other time.

JFreegman commented:
I'd prefer to keep the 12/24 time option as standard (e.g. not deprecated), and have these additions as another option which override 12/24. Just call it something like "custom_time_format". Reason being that I've never heard anyone complain about the timestamp format, so this would end up complicating things for most users while benefiting a small minority.

Also in the man page it should specify that you need to put double-quotes around the custom format strings.

> Log filename can't have '/' on *nix, or '\' on windows I imagine. What could be a sane way to deal with this?

We don't need to worry about Windows for the time being since Toxic only supports unix-like systems. But don't worry about log file names. They don't need to match custom timestamps.

louipc commented:
On Fri 20 Feb 2015 18:00 -0800, JFreegman wrote:
> I'd prefer to keep the 12/24 time option as standard (e.g. not deprecated),
> and have these additions as another option which override 12/24. Just call it
> something like "custom_time_format". Reason being that I've never heard
> anyone complain about the timestamp format, so this would end up complicating
> things for most users while benefiting a small minority.

The idea of these patches is to bring toxic more in line with a standard kind
of ncurses interface that is generally expected. Setting time format as an
integer is kind of unusual/nonstandard.

Have a look at how you set the format for these clients:

https://weechat.org/files/doc/stable/weechat_user.en.html
search for time_format, there are multiple settings all using strftime format

and
http://irssi.org/documentation/settings#a_c
 
Anyways, I've removed any sign of deprecating, and mentioned time_format before
the new options in the man page. I removed its representation internally
though, because it is only used in the setup, and the actual format strings are
used afterwards.

> Also in the man page it should specify that you need to put double-quotes
> around the custom format strings.

Done.

> > Log filename can't have '/' on *nix, or '\' on windows I imagine. What
> > could be a sane way to deal with this?
> 
> We don't need to worry about Windows for the time being since Toxic only
> supports unix-like systems. But don't worry about log file names. They don't
> need to match custom timestamps.

Yeah I don't change the filename anymore. That can be tackled some other time
maybe. Hope this looks OK now.

louipc commented:
On Mon 23 Feb 2015 19:38 -0800, JFreegman wrote:
> Merged #303.
 
Thanks. There is an issue with a long timestamp though that I can't seem to
figure out. I will describe it in a bug report.
